test/lib/checkevents.go

```diff
@@ -0,0 +1,243 @@
+}
+
+// Functions used for getting data from the REST api of the recordevents pod.
+// The interface exists for use with unit tests of this module.
+type eventGetterInterface interface {
+	getMinMax() (minRet int, maxRet int, errRet error)
+	getEntry(seqno int) (EventInfo, error)
+	trimThrough(seqno int) error
+	cleanup()
+}
+
+// Internal function to create an event store.  This is called directly by unit tests of
+// this module.
+func newTestableEventInfoStore(egi eventGetterInterface, retryInterval time.Duration,
+	timeout time.Duration) *EventInfoStore {
+	if timeout == -1 {
+		timeout = timeoutEvRetry
+	}
+	if retryInterval == -1 {
+		retryInterval = minEvRetryInterval
+	}
+	ei := &EventInfoStore{getter: egi, firstID: 1, timeout: timeout, retryInterval: retryInterval}
+	ei.start()
+	return ei
+}
+
+// Creates an EventInfoStore that is used to iteratively download events recorded by the
+// recordevents pod.  Calling this forwards the recordevents port to the local machine
+// and blocks waiting to connect to that pod.  Fails if it cannot connect within
+// the expected timeout (4 minutes currently)
+func (client *Client) NewEventInfoStore(podName string, logf logging.FormatLogger) (*EventInfoStore, error) {
+	egi, err := newEventGetter(podName, client, logf)
+	if err != nil {
+		return nil, err
+	}
+	ei := newTestableEventInfoStore(egi, -1, -1)
+	return ei, nil
+}
+
+// Starts the single threaded background goroutine used to update local state
+// from the remote REST API.
+func (ei *EventInfoStore) start() {
+	ei.closeCh = make(chan struct{})
+	ei.doRefresh = make(chan chan error)
+	go func() {
+		for {
+			select {
+			case <-ei.closeCh:
+				ei.getter.cleanup()
+				return
+			case replyCh := <-ei.doRefresh:
+				replyCh <- ei.doRetrieveData()
+			}
+		}
+	}()
+}
+
+// The data update thread used by the single threaded background goroutine
+// for updating data from the REST api.
+func (ei *EventInfoStore) doRetrieveData() error {
+	min, max, err := ei.getter.getMinMax()
+	if err != nil {
+		return fmt.Errorf("error getting MinMax %v", err)
+	}
+	ei.lock.Lock()
+	curMin := ei.firstID
+	curMax := curMin + len(ei.allEvents) - 1
+	ei.lock.Unlock()
+	if min == max+1 {
+		// Nothing to read or trim
+		return nil
+	} else {
+		if min > curMax+1 {
+			return fmt.Errorf("mismatched stored max/available min: %d, %d", curMax, min)
+		}
+		min = curMax + 1
+		// We may have data to read, definitely have data to trim.
+	}
+	var newEvents []EventInfo
+	for i := min; i <= max; i++ {
+		e, err := ei.getter.getEntry(i)
+		if err != nil {
+			return fmt.Errorf("error calling getEntry of %d %v", i, err)
+		}
+
+		newEvents = append(newEvents, e)
+	}
+	ei.lock.Lock()
+	ei.allEvents = append(ei.allEvents, newEvents...)
+	ei.lock.Unlock()
+	err = ei.getter.trimThrough(max)
+	return err
+
+}
+
+// Clean up any background resources used by the store.  Must be called exactly once after
+// the last use.
+func (ei *EventInfoStore) Cleanup() error {
```

Comment:
nit: does this needs to return something?

Reply:
removed the unused return